sigstore-java/src/main/java/dev/sigstore/KeylessVerifier.java

```diff
@@ -399,46 +405,110 @@ private void checkDsseEnvelope(
+      if (algorithm != PayloadHash.Algorithm.SHA_256) {
+        throw new KeylessVerificationException(
+            "Cannot process DSSE entry with hashing algorithm " + algorithm.toString());
+      }
 
-    byte[] payloadDigest;
-    try {
-      payloadDigest = Hex.decode(rekorDsse.getPayloadHash().getValue());
-    } catch (DecoderException de) {
-      throw new KeylessVerificationException(
-          "Could not decode hex sha256 artifact hash in hashrekord", de);
-    }
+      // check if the digest over the dsse payload matches the digest in the transparency log entry
+      byte[] payloadDigest;
+      try {
+        payloadDigest = Hex.decode(rekorDsse.getPayloadHash().getValue());
+      } catch (DecoderException de) {
+        throw new KeylessVerificationException(
+            "Could not decode hex sha256 artifact hash in hashrekord", de);
+      }
 
-    byte[] calculatedDigest = Hashing.sha256().hashBytes(dsseEnvelope.getPayload()).asBytes();
-    if (!Arrays.equals(calculatedDigest, payloadDigest)) {
-      throw new KeylessVerificationException(
-          "Digest of DSSE payload in bundle does not match DSSE payload digest in log entry");
-    }
+      byte[] calculatedDigest = Hashing.sha256().hashBytes(dsseEnvelope.getPayload()).asBytes();
+      if (!Arrays.equals(calculatedDigest, payloadDigest)) {
+        throw new KeylessVerificationException(
+            "Digest of DSSE payload in bundle does not match DSSE payload digest in log entry");
+      }
 
-    // check if the signature over the dsse payload matches the signature in the rekorEntry
-    if (rekorDsse.getSignatures().size() != 1) {
-      throw new KeylessVerificationException(
-          "DSSE log entry must have exactly 1 signature, but found: "
-              + rekorDsse.getSignatures().size());
-    }
+      // check if the signature over the dsse payload matches the signature in the rekorEntry
+      if (rekorDsse.getSignatures().size() != 1) {
+        throw new KeylessVerificationException(
+            "DSSE log entry must have exactly 1 signature, but found: "
+                + rekorDsse.getSignatures().size());
+      }
 
-    if (!Base64.getEncoder()
-        .encodeToString(dsseEnvelope.getSignature())
-        .equals(rekorDsse.getSignatures().get(0).getSignature())) {
-      throw new KeylessVerificationException(
-          "Provided DSSE signature materials are inconsistent with DSSE log entry");
+      if (!Base64.getEncoder()
+          .encodeToString(dsseEnvelope.getSignature())
+          .equals(rekorDsse.getSignatures().get(0).getSignature())) {
+        throw new KeylessVerificationException(
+            "Provided DSSE signature materials are inconsistent with DSSE log entry");
+      }
+    } else if ("0.0.2".equals(version)) {
+      DSSELogEntryV002 logEntrySpec;
+      try {
+        DSSELogEntryV002.Builder builder = DSSELogEntryV002.newBuilder();
+        ProtoJson.parser()
+            .ignoringUnknownFields()
```

Comment:
this `ignoringUnknownFields()` should have already been applied by our ProtoJson.parser() proxy